dev-libs/date/date-3.0.1.ebuild

```
@@ -0,0 +1,35 @@
+# Copyright 1999-2024 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+inherit cmake
+
+DESCRIPTION="A date and time library based on the C++11/14/17 <chrono> header"
+HOMEPAGE="https://github.com/HowardHinnant/date"
+SRC_URI="https://github.com/HowardHinnant/${PN}/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="MIT"
+SLOT="0/${PV}"
+KEYWORDS="~amd64"
+IUSE="only-c-locale test"
+RESTRICT="!test? ( test )"
+
+src_prepare() {
+	cmake_src_prepare
+}
+
+src_configure() {
+	local mycmakeargs=(
+		-DBUILD_TZ_LIB=ON
+		-DUSE_SYSTEM_TZ_DB=ON
+		-DENABLE_DATE_TESTING=$(usex test)
+		-DCOMPILE_WITH_C_LOCALE=$(usex only-c-locale)
+	)
+	cmake_src_configure
+}
+
+src_test() {
+	cd "${BUILD_DIR}" || die
+	ninja testit
```

Without proper error handling, you get failing tests but the phase passes making it look like all tests ran fine.
```
95% tests passed, 5 tests failed out of 108

Total Test time (real) =   1.50 sec

The following tests FAILED:
         38 - date_test_pass_parse_test (Subprocess aborted)
        100 - posix_pass_ptz_test (Subprocess aborted)
        101 - solar_hijri_test_pass_parse_test (Subprocess aborted)
        105 - tz_test_pass_validate_test (Subprocess aborted)
        108 - tz_test_pass_zoned_time_deduction_test (Subprocess aborted)
Errors while running CTest
Output from these tests are in: /var/tmp/portage/dev-libs/date-3.0.1/work/date-3.0.1_build/Testing/Temporary/LastTest.log
Use "--rerun-failed --output-on-failure" to re-run the failed cases verbosely.
ninja: build stopped: subcommand failed.
```

Add
```suggestion
    ninja testit || die
```
and try yourself what happens :)

I didn't try 3.0.0 but I saw the previous ebuild had some... workarounds for failing tests, which may be related to this.

---

Fyi tests might still need some working, first of all:
```
Results for /var/tmp/portage/dev-libs/date-3.0.1/work/date-3.0.1/test:
./testit: line 162: clang++: command not found
using 
with -std=c++14  -I/var/tmp/portage/dev-libs/date-3.0.1/work/date-3.0.1 -Wall /var/tmp/portage/dev-libs/date-3.0.1/work/date-3.0.1/src/tz.cpp -lcurl -I/var/tmp/portage/dev-libs/date-3.0.1/work/date-3.0.1/include -I/var/tmp/portage/dev-libs/date-3.0.1/work/date-3.0.1/include/date 
----------------------------------------------------
sections without tests   : 0
sections with failures   : 9
sections without failures: 0
                       +   ----
total number of sections : 9
----------------------------------------------------
number of tests failed   : 71
number of tests passed   : 37
                       +   ----
total number of tests    : 108
****************************************************
 * ERROR: dev-libs/date-3.0.1::gentoo failed (test phase):
 *   (no error message)
```
looks like you need clang for whatever reason.

even after installing clang, a bunch of tests fail because "timezone data" isn't found.

---

> What should I do? Do I just restrict the tests?

couldn't you patch out that failing test(s) and document the test failure? in java we typically have [this kind of patches](https://gitweb.gentoo.org/repo/gentoo.git/tree/dev-java/bsh/files/bsh-2.1.1-skip-tests.patch) with bug reports blocking a certain [tracker bug](https://bugs.gentoo.org/856409).

---

> That's a great idea!

If not too long i would put the failure protocol on top of the patch, like [here](https://github.com/gentoo/gentoo/pull/33579/commits/3066db1868f80fb95b8dc1fae9c682597c8259c1#diff-2227b00bbcba13c7f07a0090dcd60d479b0f9d16ea1ca05f1d73d0f9ffaa6975), as an explanation why those tests are skipped.